x-pack/plugins/alerting/server/lib/errors/es_error_parser.ts

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+// import { ResponseError } from '@elastic/elasticsearch/lib/errors';
+import { ElasticsearchError } from './';
+
+interface ErrorObject {
+  caused_by?: {
+    reason?: string;
+    caused_by?: {};
+  };
+  failed_shards?: Array<{
+    reason?: {
+      caused_by?: {
+        reason?: string;
+      };
+    };
+  }>;
+}
+
+const getEsCause = (obj: ErrorObject = {}, causes: string[] = []): string[] => {
+  const updated = [...causes];
+
+  if (obj.caused_by) {
+    if (obj.caused_by?.reason) {
+      updated.push(obj.caused_by?.reason);
+    }
+
+    // Recursively find all the "caused by" reasons
+    return getEsCause(obj.caused_by, updated);
```

Comment:
nit: It looks like sometimes the recursively nested `caused_by` reasons can be pretty redundant. Could we possibly dedupe them?